src/librustdoc/json/conversions.rs

```diff
@@ -634,42 +634,28 @@ impl FromWithTcx<clean::Enum> for Enum {
         Enum {
             generics: generics.into_tcx(tcx),
             variants_stripped,
-            variants: ids(variants, tcx),
+            variants: ids(&variants, tcx),
             impls: Vec::new(), // Added in JsonRenderer::item
         }
     }
 }
 
-impl FromWithTcx<clean::VariantStruct> for Struct {
-    fn from_tcx(struct_: clean::VariantStruct, tcx: TyCtxt<'_>) -> Self {
-        let fields_stripped = struct_.has_stripped_entries();
-        let clean::VariantStruct { struct_type, fields } = struct_;
-        Struct {
-            struct_type: from_ctor_kind(struct_type),
-            generics: Generics { params: vec![], where_predicates: vec![] },
-            fields_stripped,
-            fields: ids(fields, tcx),
-            impls: Vec::new(),
-        }
-    }
-}
-
 impl FromWithTcx<clean::Variant> for Variant {
     fn from_tcx(variant: clean::Variant, tcx: TyCtxt<'_>) -> Self {
         use clean::Variant::*;
+
         match variant {
-            CLike => Variant::Plain,
-            Tuple(fields) => Variant::Tuple(
-                fields
-                    .into_iter()
-                    .filter_map(|f| match *f.kind {
-                        clean::StructFieldItem(ty) => Some(ty.into_tcx(tcx)),
-                        clean::StrippedItem(_) => None,
-                        _ => unreachable!(),
-                    })
-                    .collect(),
-            ),
-            Struct(s) => Variant::Struct(ids(s.fields, tcx)),
+            CLike => Variant { kind: StructKind::Unit, fields: Vec::new(), fields_stripped: false },
+            Tuple(fields) => Variant {
+                kind: StructKind::Tuple,
+                fields: ids(&fields, tcx),
+                fields_stripped: fields.iter().any(|i| i.is_stripped()),
+            },
+            Struct(s) => Variant {
+                kind: StructKind::NamedFields,
```

Comment:
Just thought about it: `NamedFields` sounds a bit weird. Wouldn't it be better to have something similar to what already exists? For example [VariantData](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_hir/hir/enum.VariantData.html).

Reply:
The problem with that is having `struct.kind == StructKind::Struct` isn't informative, as all 3 kinds can be "structs".

Another option is to use `StructKind::Normal`, as most structs use `{}`, but thats not the "normal" option for enum variants, which are mostly tuple or unit.

While it's not the name used in compiler internals, we haven't done that for other parts of the output (eg using `type_` instead of `ty`.

Reply:
Multiplying the number of terms isn't the best idea either but I'm not great for naming... Does it seem good to you @Manishearth ?